src/main/java/core/basesyntax/db/FruitDaoImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitDaoImpl implements FruitDao {
+    private final Map<String, Integer> fruitDB;
+
+    public FruitDaoImpl() {
+        fruitDB = new HashMap<>();
+    }
+
+    @Override
+    public Integer putFruits(String key, Integer value) {
+        Integer numberOfFruits = fruitDB.get(key);
+
+        validateValue(numberOfFruits, value);
+        if (numberOfFruits == null) {
+            return fruitDB.put(key, value);
+        }
+        return fruitDB.put(key, numberOfFruits + value);
+    }
+
+    public Map<String, Integer> getFruitDB() {
+        return new HashMap<>(fruitDB);
+    }
+
+    private void validateValue(Integer numberOfFruits, Integer value) {
+        if (numberOfFruits == null && value < 0
+                || numberOfFruits != null && numberOfFruits + value < 0) {
+            throw new RuntimeException("Fruit stock cannot be less than 0");
+        }
```

Comment:
The validation logic in `validateValue` method should not allow the `value` to be negative when `numberOfFruits` is null. This is good, but it should also be checked during the `putFruits` method, before updating the storage. This relates to the checklist item: Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.